src/machine/machined-dbus.c

```diff
@@ -166,7 +166,7 @@ static int method_get_machine_by_pid(sd_bus_message *message, void *userdata, sd
                         return r;
         }
 
-        r = manager_get_machine_by_pid(m, pid, &machine);
+        r = manager_get_machine_by_pidref(m, &PIDREF_MAKE_FROM_PID(pid), &machine);
```

Comment:
similar, as above. there's a helper bus_creds_get_pidref() aleady, which you can use instead of sd_bus_creds_get_pid() above.

Note you must then add SD_BUS_CREDS_PIDFD to the the mask on sd_bus_query_sender_creds() for this to work.